### DISCUSSION TITLE
GPIO15 on boot

### DISCUSSION THREAD

**Discussion by host-felice** [arendst/Tasmota#17531] [Q&A]
Everything started from here: https://github.com/arendst/Tasmota/discussions/17523

I am not sure if it is related to that problem, but I noticed that powering on my shellys 2.5 and 1PM, with tasmota, they shortly turn on the relay on GPIO15. It doesn't happen on restart. With the original software, they don't "click" the relay.

I never paid attention to it, I thought it was normal since GPIO15 is not the best gpio to use (internal pull down and pulled low on boot), but I don't see a reason to behave like this. Moreover, it doesn't happen with the original shelly firmware.

If with the 1PM could be not a real problem, the 2.5 has 2 outputs. If the load on the GPIO15 relay is short, the circuit breaker shuts down. You cannot turn the shelly on to use at least the other relay with a working load because the GPIO15 relay is still short and it "clicks" at power on -> the circuit breaker shuts down the whole line. If you have something else than the shelly on that circuit breaker, everything is not usable unless you repair the short or you unplug the shelly.

I tried with a wemos d1 mini: while powering on, a relay connected to GPIO15 (D8) does the same with both high and low trigger relays.

If the original shelly software doesn't do it, it is probably something that can be fixed in tasmota too.

Thanks and sorry for the long post.

**Comment by arendst** [comment]
Thnx for bringing this up.

It used to be solved a long time ago by setting pins to input on restart (and power up). I see, for some reason, this feature is disabled. If I enable it the relays stop being turned on briefly.

I need further research why we removed this functionality. Might have had something to do with JTAG debugging which doesn't like GPIO's being set to input.....

**Comment by host-felice** [comment]
Thanks to your great work :-) Waiting for your news

**Comment by arendst** [comment]
Fixed in latest dev. (https://github.com/arendst/Tasmota/commit/4ee6aee35e88a5e2aa4225442091036bad82eb62)

**Reply by sfromis** [reply]
I also saw this behavior, now no click from the gpio 15 Relay2 when applying power to a Shelly 2.5.

**Comment by host-felice** [comment]
Great!
But it didn't solve the https://github.com/arendst/Tasmota/discussions/17523

It is incredible! Tasmota works @3,3V with the serial adapter, it doesn't @220V.

With the shelly firmware, my 1PM works both @3,3V and @220V. It is the only one acting this way. Other 1PMs work without problem.

**Reply by sfromis** [reply]
Whatever happens with 3.3V vs 220V should be unrelated to this thread.

**Comment by bholmesdev** [comment]


**Reply by sfromis** [reply]
You were not mentioned in this thread.

**Reply by barbudor** [reply]
@bholmesdev only you can unsubscribe from the thread
We faced some spamming earlier today and a lot of emails  have been brought in.
We have no way of removing you. Only you can

**Reply by sfromis** [reply]
You may also have set up a watch filter, choosing to be notified for all discussions. Check the "Watch" drop down near the top right, and uncheck if relevant.

**Reply by bholmesdev** [reply]
Ah got it! Didn't know if I'd been pinging across multiple discussions, sorry 👍

**Comment by host-felice** [comment]
what are you talking about?
